File: 26주차/이연상/4와 7.py
# [BOJ] 4와 7

import sys
N = int(sys.stdin.readline())
result =''
while N > 0:
    m = N % 2 
    N = N // 2 
    if m == 0: 
        N -= 1 
        result = '7'+ result
    else: #홀수면
        result = '4'+ result 
print(result)

File: 55주차/이연상/4와 7.java
// [BOJ] 4와 7

import java.util.Scanner;

public class FourAndSeven {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        StringBuilder result = new StringBuilder();

        while (N > 0) {
            int m = N % 2;
            N = N / 2;
            if (m == 0) {
                N -= 1;
                result.insert(0, '7');
            } else { // 홀수면
                result.insert(0, '4');
            }
        }

        System.out.println(result);
    }
}